pep-lint.py

```diff
@@ -0,0 +1,427 @@
+            continue
+        yield from check_direct_links(line_num, line.lstrip())
+
+
+def check_headers(lines):
+    yield from _validate_pep_number(next(iter(lines), ""))
+
+    found_headers = {}
+    line_num = 0
+    for line_num, line in enumerate(lines, start=1):
+        if line.strip() == "":
+            headers_end_line_num = line_num
+            break
+        if match := HEADER_PATTERN.match(line):
+            header = match[1]
+            if header in ALL_HEADERS:
+                if header not in found_headers:
+                    found_headers[match[1]] = line_num
+                else:
+                    yield line_num, f"Must not have duplicate header: {header} "
+            else:
+                yield line_num, f"Must not have invalid header: {header}"
+    else:
+        headers_end_line_num = line_num
+
+    yield from _validate_required_headers(found_headers)
+
+    shifted_line_nums = list(found_headers.values())[1:]
+    for i, (header, line_num) in enumerate(found_headers.items()):
+        start = line_num - 1
+        end = headers_end_line_num - 1
+        if i < len(found_headers) - 1:
+            end = shifted_line_nums[i] - 1
+        remainder = "\n".join(lines[start:end]).removeprefix(f"{header}:")
+        if remainder != "":
+            if remainder[0] not in {" ", "\n"}:
+                yield line_num, f"Headers must have a space after the colon: {header}"
+            remainder = remainder.lstrip()
+
+        if header == "Title":
+            yield from _validate_title(line_num, remainder)
+        elif header == "Author":
+            yield from _validate_author(line_num, remainder)
+        elif header == "Sponsor":
+            yield from _validate_sponsor(line_num, remainder)
+        elif header in {"BDFL-Delegate", "PEP-Delegate"}:
+            yield from _validate_delegate(line_num, remainder)
+        elif header == "Discussions-To":
+            yield from _validate_discussions_to(line_num, remainder)
+        elif header == "Status":
+            yield from _validate_status(line_num, remainder)
+        elif header == "Type":
+            yield from _validate_type(line_num, remainder)
+        elif header == "Topic":
+            yield from _validate_topic(line_num, remainder)
+        elif header == "Content-Type":
+            yield from _validate_content_type(line_num, remainder)
+        elif header in {"Requires", "Replaces", "Superseded-By"}:
+            yield from _validate_pep_references(line_num, remainder)
+        elif header == "Created":
+            yield from _validate_created(line_num, remainder)
+        elif header == "Python-Version":
+            yield from _validate_python_version(line_num, remainder)
+        elif header == "Post-History":
+            yield from _validate_post_history(line_num, remainder)
+        elif header == "Resolution":
+            yield from _validate_resolution(line_num, remainder)
```

Comment:
I considered this initially, but I found it quite useful to be able to avoid the indirection of a dispatch dictionary and directly jump to a function via my IDE. We could split this chunk out into a dedicated function, though?